src/main/kotlin/org/ionproject/integration/extractor/implementation/ITextPdfExtractor.kt

```diff
@@ -0,0 +1,42 @@
+package org.ionproject.integration.extractor.implementation
+
+import com.itextpdf.kernel.pdf.PdfDocument
+import com.itextpdf.kernel.pdf.PdfReader
+import com.itextpdf.kernel.pdf.canvas.parser.PdfTextExtractor
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.ionproject.integration.extractor.`interface`.PdfExtractor
+import org.ionproject.integration.extractor.exception.PdfExtractorException
+import org.ionproject.integration.utils.Try
+
+class ITextPdfExtractor : PdfExtractor {
+    /**
+     * Extract text data from pdf file locate at [pdfPath]
+     * @return Try<MutableList<String>>
+     *     Success - String list contains all extracted data in text format
+     *     Failure - PdfExtractorException
+     */
+    override suspend fun extract(pdfPath: String): Try<MutableList<String>> {
+        if (pdfPath.isEmpty()) return Try.ofError(PdfExtractorException("Empty path"))
+
+        val tryReader = Try.of { PdfReader(pdfPath) }
+        if (tryReader is Try.Error) return Try.ofError(PdfExtractorException("File doesn't exist or Invalid file format"))
```

Comment:
We should need to do these explicit `ifs` when using `Try`. Did you face any blocking issues?

Reply:
`PdfReader(pdfPath)` throws an `java.io.IOException`, but we want all errors to be of type `PdfExtractorException` for easier error handling.

Reply:
We can add a `mapError` to the `Try` to do this mapping.